pm/queue.go

```diff
@@ -136,19 +125,25 @@ func (q *ticketQueue) startQueueLoop() {
 			}
 		}
 
-		nextTicket := q.queue[0]
-
 		// Unlock since the queue is non-empty now
 		q.cond.L.Unlock()
 
 		select {
-		case maxFloat := <-q.maxFloatUpdate:
-			if q.sufficientMaxFloat(maxFloat, nextTicket) {
-				select {
-				case q.redeemable <- nextTicket:
-					q.removeHead()
-				case <-q.quit:
-					return
+		case err := <-sub.Err():
+			glog.Error(err)
+		case latestBlock := <-blockNums:
+			numTickets := len(q.queue)
+			for i := 0; i < numTickets; i++ {
+				nextTicket := q.queue[0]
```

Comment:
I also expected N to be quite small and therefore assumed just looping through might be the best trade-off to make considering end-of-round optimization and that we really want to try and redeem all current tickets in the queue before the round ends. 

Good point about thread safety.

Reply:
On second thought, I removed this test because we don't actually signal block number updates concurrently like we did with max float signalling. 

I should keep testing concurrent adds though.

Reply:
fixed in 756fe853cb627a508e1f33e21725a8fc7b465805